contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -142,15 +140,27 @@ library SafeERC20 {
      * on the return value: the return value is optional (but if data is returned, it must not be false).
      * @param token The token targeted by the call.
      * @param data The call data (encoded using abi.encode or one of its variants).
+     *
+     * This is a variant of {_callOptionalReturnBool} that reverts if call fails to meet the requirements.
      */
     function _callOptionalReturn(IERC20 token, bytes memory data) private {
-        // We need to perform a low level call here, to bypass Solidity's return data size checking mechanism, since
-        // we're implementing it ourselves. We use {Address-functionCall} to perform this call, which verifies that
-        // the target address contains contract code and also asserts for success in the low-level call.
-
-        bytes memory returndata = address(token).functionCall(data);
-        if (returndata.length != 0 && !abi.decode(returndata, (bool))) {
-            revert SafeERC20FailedOperation(address(token));
+        uint256 returnSize;
+        uint256 returnValue;
+        assembly ("memory-safe") {
+            let success := call(not(0), token, 0, add(data, 0x20), mload(data), 0, 0x20)
+            // bubble errors
+            if iszero(success) {
+                let ptr := mload(0x40)
+                returndatacopy(ptr, 0, returndatasize())
+                revert(ptr, returndatasize())
+            }
+            returnSize := returndatasize()
+            returnValue := mload(0)
+        }
+        if (returnSize == 0) {
+            if (address(token).code.length == 0) revert Address.AddressEmptyCode(address(token));
+        } else {
+            if (returnValue == 0) revert SafeERC20FailedOperation(address(token));
```

Comment:
We need to clarify this error change in the CHANGELOG, we've documented others over there.

Reply:
Its not an error change. 

This PR does not change the tests, and both line are covered ... this means that this if is REQUIRED to replicate the errors generation from 5.0

Reply:
Alternativelly, we can decide to only emit the `SafeERC20FailedOperation` error in both cases, but that would be a change.

Reply:
I pushed a commit with that change. Solving this for now